### PR TITLE
[MRG+1] Clean up BoundaryNorm docstring

### PR DESCRIPTION
**Pull request by dstansby**
- Make numpydoc compatible
- Add note that calling `invert` will always raise a `ValueError`

### CODE REVIEW

File: lib/matplotlib/colors.py [hunk 1]
```diff
@@ -1247,25 +1247,27 @@ class BoundaryNorm(Normalize):
...
+
+            If clip is *False*, out of range values are mapped to *-1* if they
+            are below *boundaries[0]* or mapped to *ncolors* if they are
+            above *boundaries[-1]*. These are then converted to valid indices
+            by :meth:`Colormap.__call__`.
 
-        then v is mapped to color j;
+        Notes
+        -----
+        If :code:`b[i] <= v < b[i+1]` then v is mapped to color j;
```
Comment (anntzer) [inline lib/matplotlib/colors.py]: What happens if the two lengths don't match?

File: lib/matplotlib/colors.py [hunk 2]
```diff
@@ -1247,25 +1247,27 @@ class BoundaryNorm(Normalize):
...
-
-            b[i] <= v < b[i+1]
+        Parameters
+        ----------
+        boundaries : sequence
+            Monotonically increasing sequence of boundaries
+        ncolors : int
+            Number of colors in the colormap to be used
+        clip : bool, optional
+            If clip is *True*, out of range values are mapped to *0* if they
```
Comment (anntzer) [inline lib/matplotlib/colors.py]: Use double backquotes here and below.

Comment (dstansby) [inline lib/matplotlib/colors.py]: For 'clip' or for '*True* and '*0*'?

File: lib/matplotlib/colors.py (hunk 1 above)
Comment (dstansby) [inline lib/matplotlib/colors.py]: I have no idea, will try and work out and update this later today.

File: lib/matplotlib/colors.py (hunk 2 above)
Comment (anntzer) [inline lib/matplotlib/colors.py]: I would say for all (they're all "code").

Comment (dstansby) [inline lib/matplotlib/colors.py]: But http://matplotlib.org/devdocs/devel/documenting_mpl.html#formatting says to explicitly not do that! (at least for arguments to functions)

Comment (anntzer) [inline lib/matplotlib/colors.py]: Indeed, that simply means I've been doing things wrong :-)
https://docs.python.org/devguide/documenting.html#id3 (the mpl docs refer to consistency with python itself) says to use emphasis specifically for local variables (i.e. arguments) only.

File: lib/matplotlib/colors.py [hunk 3]
```diff
@@ -1246,26 +1246,28 @@ class BoundaryNorm(Normalize):
...
-        Out-of-range values are mapped
-        to -1 if low and ncolors if high; these are converted
-        to valid indices by
-        :meth:`Colormap.__call__` .
-        If clip == True, out-of-range values
-        are mapped to 0 if low and ncolors-1 if high.
+        Notes
+        -----
+        If :code:`b[i] <= v < b[i+1]` then v is mapped to
+        floor( :math:`\frac{n_{colors}- 1}{n_{bins} - 1} * i`), where nbins is
```
Comment (phobson) [inline lib/matplotlib/colors.py]: should the floor function be in the math directive?

Comment (dstansby) [inline lib/matplotlib/colors.py]: I thought it looked better visually outside, but I can put it inside.

Comment (dstansby) [inline lib/matplotlib/colors.py]: Actually this is about to go in my next revision anyway

File: lib/matplotlib/colors.py [hunk 4]
```diff
@@ -1246,17 +1246,17 @@ class BoundaryNorm(Normalize):
...
             Number of colors in the colormap to be used
         clip : bool, optional
-            If clip is *True*, out of range values are mapped to *0* if they
-            are below *boundaries[0]* or mapped to *ncolors - 1* if they are
-            above *boundaries[-1]*.
+            If *clip* is ``True``, out of range values are mapped to 0 if they
+            are below ``boundaries[0]`` or mapped to *ncolors* - 1 if they are
+            above ``boundaries[-1]``.
 
             If clip is *False*, out of range values are mapped to *-1* if they
```
Comment (phobson) [inline lib/matplotlib/colors.py]: ``False`` and ``boundaries[0]`` to be consistent with the previous paragrpha

File: lib/matplotlib/colors.py [hunk 5]
```diff
@@ -1247,25 +1247,29 @@ class BoundaryNorm(Normalize):
...
-            a monotonically increasing sequence
-        *ncolors*
-            number of colors in the colormap to be used
-
-        If::
-
-            b[i] <= v < b[i+1]
+        Parameters
+        ----------
+        boundaries : sequence
```
Comment (NelleV) [inline lib/matplotlib/colors.py]: this should be array-like

File: lib/matplotlib/colors.py [hunk 6]
```diff
@@ -1247,25 +1247,29 @@ class BoundaryNorm(Normalize):
...
+            Monotonically increasing sequence of boundaries
+        ncolors : int
+            Number of colors in the colormap to be used
+        clip : bool, optional
+            If *clip* is ``True``, out of range values are mapped to 0 if they
```
Comment (NelleV) [inline lib/matplotlib/colors.py]: Can you please remove the *? They make docstrings hard to read in the terminal

File: lib/matplotlib/colors.py [hunk 7]
```diff
@@ -1304,6 +1308,13 @@ def __call__(self, value, clip=None):
         return ret
 
     def inverse(self, value):
+        """
+        Raises
+        ------
+        ValueError
+            BoundaryNorm is not invertible, so calling this method will always
```
Comment (NelleV) [inline lib/matplotlib/colors.py]: :+1:

**Comment by dstansby** [comment]
Those should be fixed now

File: lib/matplotlib/colors.py [hunk 8]
```diff
@@ -1247,25 +1247,29 @@ class BoundaryNorm(Normalize):
...
+        is mapped to the color with the same index.
 
-        Out-of-range values are mapped
-        to -1 if low and ncolors if high; these are converted
-        to valid indices by
-        :meth:`Colormap.__call__` .
-        If clip == True, out-of-range values
-        are mapped to 0 if low and ncolors-1 if high.
+        If the number of bins doesn't equal *ncolors*, the color is chosen
+        by linear inpolation of the bin number onto color numbers.
```
Comment (anntzer) [inline lib/matplotlib/colors.py]: "interpolation"

**Comment by anntzer** [comment]
Thanks!

**Pull request merged**